### ISSUE TITLE
plugin-replace: Can the replacement occur on the AST instead of as a string?

### ISSUE THREAD

**Issue by jespertheend** [rollup/plugins#803]
- Rollup Plugin Name: plugin-replace
- Rollup Plugin Version: 2.3.4

### Feature Use Case
I'd like to replace `DEBUG` with `true` or `ENV` with `"DEBUG"` for instance. But this will also replace these words on locations where it's not desired.

For example`if(DEBUG) { console.log("DEBUG mode enabled") }` will become `if(true) { console.log("true mode enabled") }`

Or an example from https://github.com/rollup/rollup-plugin-replace/issues/33:
`Vue.prototype.ASSET_URL = ASSET_URL;` becoming `Vue.prototype.'http://my-asset-url/' = 'http://my-asset-url/';` and causing a syntax error.

### Feature Proposal
Perhaps an extra flag in the options argument for the `replace()` plugin could work. One which, when set to true, will replace in the AST instead of as a string.

This feature was also requested at https://github.com/rollup/rollup-plugin-replace/issues/33 a while back but I couln't find anything related to this.

**Comment by jespertheend** [comment]
As an alternative, https://www.npmjs.com/package/rollup-plugin-define seems to get the job done as well.

**Comment by shellscape** [comment]
The simple solution is to name your assignment slightly differently. It'd be a heavy lift to use the ast to replace, and while I'm sure we'd consider the contribution, it's unlikely to get development from maintainers.

**Comment by jespertheend** [comment]
That makes sense. rollup-plugin-define seems like a good enough solution for my use case at the moment.

**Comment by shellscape** [comment]
This should be resolved by #798

**Comment by jespertheend** [comment]
Oh neat, I wasn't aware of this. Thanks!
Does this also prevent replaces in strings? For example:
```js
if(DEBUG) {
    console.log("DEBUG mode enabled");
}
```
becoming
```js
if(true) {
    console.log("true mode enabled");
}
```
